javascript/ql/lib/semmle/javascript/frameworks/data/internal/ApiGraphModelsSpecific.qll

```diff
@@ -31,6 +31,15 @@ import semmle.javascript.frameworks.data.internal.AccessPathSyntax as AccessPath
 import JS::DataFlow as DataFlow
 private import AccessPathSyntax
 
+bindingset[rawType]
+predicate parseTypeString(string rawType, string package, string qualifiedName) {
+  exists(string regexp |
+    regexp = "('[^']+'|[^.]+)(.*)" and
+    package = rawType.regexpCapture(regexp, 1).regexpReplaceAll("^'|'$", "") and
+    qualifiedName = rawType.regexpCapture(regexp, 2).regexpReplaceAll("^\\.", "")
+  )
+}
+
```

Comment:
I understand this, but I basically dream in regular expressions at this point. 
A documentation string could be nice.  

```suggestion
/**
 * Holds if `rawType` represents the JavaScript npm `package` with type `qualifiedName`.  
 * The `qualifiedName` is everything after the first ".".   
 * So e.g. `rawType = "foo.bar"` will parse to `package = "foo" and qualifiedName = "bar"`.  
 * `package´ can be enclosed in single-quotes if the package name includes dots, 
 * e.g. `'my.package'.type` parses to `package = "my.package" and qualifiedName = "type"`.  
 */ 
bindingset[rawType]
predicate parseTypeString(string rawType, string package, string qualifiedName) {
  exists(string regexp |
    regexp = "('[^']+'|[^.]+)(.*)" and
    package = rawType.regexpCapture(regexp, 1).regexpReplaceAll("^'|'$", "") and
    qualifiedName = rawType.regexpCapture(regexp, 2).regexpReplaceAll("^\\.", "")
  )
}

```

Reply:
Also, is this syntax documented anywhere?

Reply:
> Also, is this syntax documented anywhere?

Not yet, but I'll be working on documentation for the MaD format